src/Plugin/Action/SendEmail.php

```diff
@@ -0,0 +1,134 @@
+ *       required = FALSE,
+ *     ),
+ *     "language" = @ContextDefinition("language",
+ *       label = @Translation("Language"),
+ *       description = @Translation("If specified, the language used for getting the mail message and subject."),
+ *       required = FALSE,
+ *     ),
+ *   }
+ * )
+ *
+ * @todo: Define that message Context should be textarea comparing with textfield Subject
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add group information from Drupal 7.
+ */
+class SendEmail extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var LoggerInterface $logger
+   */
+  protected $logger;
+
+  /**
+   * @var MailManagerInterface $mailManager
+   */
+  protected $mailManager;
+
+  /**
+   * Constructs a SendEmail object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param LoggerInterface $logger
+   *   The alias storage service.
+   * @param $mail_manager
+   *   The alias mail manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LoggerInterface $logger, $mail_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->mailManager = $mail_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory'),
+      $container->get('plugin.manager.mail')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t('Send email');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $send_to = $this->getContextValue('send_to');
+    // @todo: Implement hook_mail_alter() in order to modify the FROM header according to https://www.drupal.org/node/2164905.
+    $reply = $this->getContextValue('reply');
+    $params = array(
+      'subject' => $this->getContextValue('subject'),
+      'message' => $this->getContextValue('message'),
+      'langcode' => "en", //$this->getContextValue('language'),
+    );
+    // Set a unique key for this mail.
+    $key = 'rules_action_mail_' . $this->getPluginId();
+
+    $message = $this->mailManager->mail('rules', $key, $send_to, $params['langcode'], $params, $reply);
+
+    if ($message['result']) {
+      $recipient = implode(", ", $send_to);
+      $this->logger->log(LogLevel::NOTICE, $this->t('Successfully sent email to %recipient', array('%recipient' => $recipient)));
+
+    }
+    return $message;
```

Comment:
Why are we returning something here? The execute() method of plugins shouldn't return anything?